akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala

```diff
@@ -12,12 +12,40 @@ import StatusCodes.ClientError
  * repeating anything present in the message itself (in order to not open holes for XSS attacks),
  * while the detail can contain additional information from any source (even the request itself).
  */
-final case class ErrorInfo(summary: String = "", detail: String = "") {
+final class ErrorInfo(
+  val summary:         String = "",
+  val detail:          String = "",
+  val errorHeaderName: String = ""
+) extends scala.Product with scala.Serializable with scala.Equals with java.io.Serializable {
   def withSummary(newSummary: String) = copy(summary = newSummary)
   def withSummaryPrepended(prefix: String) = withSummary(if (summary.isEmpty) prefix else prefix + ": " + summary)
+  def withHeaderNamePrepend(headerName: String) = setErrorHeaderName(errorHeaderName = headerName)
```

Comment:
This doesn't actually prepend the header name, right?

Reply:
Surely you are right.　rename method.